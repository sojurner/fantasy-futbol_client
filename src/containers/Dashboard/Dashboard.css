.dashboard {
  grid-column: 1 / 2;
  background-color: #f0f0f0;
  margin: 0 1rem;
  border-right: 2px solid #cacaca; }

.dash-text {
  color: #a8a8a8;
  opacity: 0.4; }

.users {
  border-bottom: 1px solid red; }
