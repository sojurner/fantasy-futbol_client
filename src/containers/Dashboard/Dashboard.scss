@font-face {
  font-family: robot;
  src: url('../../fonts/RobotoCondensed-Regular.ttf');
}

.dashboard {
  grid-column: 1 / 2;
  background-color: rgb(240, 240, 240);
  margin: 0 1rem;
  border-right: 2px solid rgb(202, 202, 202);
}

.dash-text {
  color: rgb(168, 168, 168);
  opacity: 0.4;
}

.team {
  border-bottom: 1px solid rgb(192, 191, 191);
  padding-bottom: 1rem;
}

.user-icon {
  padding: 1rem 0;
  border-bottom: 1px solid rgb(192, 191, 191);
}

.user-nav {
  display: flex;
  flex-direction: column;
}

.user,
.players,
.dashboard-players {
  text-decoration-line: none;
  padding: 0.5rem 0.7rem;
  font-family: robot;
  font-size: 1.1rem;
  color: rgb(90, 90, 90);
  letter-spacing: 0.5px;
}

.user {
  margin: 0 0 0 1.1rem;
  width: 60%;
  font-size: 1.3rem;
  letter-spacing: 1.2px;
}

.user:hover {
  background-color: #e2e0e0;
}

.user-option {
  padding: 1rem 0;
  display: flex;
}

.active {
  border-left: 2px solid #3e7fdf;
  background-color: rgb(235, 232, 232);
}

.fa-plus-circle {
  color: rgb(92, 127, 221);
  cursor: pointer;
  margin-right: 1rem;
  font-size: 1.5rem;
  margin-top: 0.2rem;
}

.fa-plus-circle:hover {
  color: rgb(62, 97, 192);
  animation: pulse 1s linear infinite;
  border-radius: 50%;
  border-color: #3e7fdf;
}

.user-icon {
  display: flex;
  justify-content: space-between;
  width: 100%;
}

.users {
  margin: 0 0 0 1rem;
}

.fa-minus-circle {
  align-self: center;
  font-size: 1.5rem;
}

.username-input {
  height: 2rem;
  width: 10rem;
  margin: auto;
}

@keyframes pulse {
  0% {
    box-shadow: 0 0 0 0 rgba(23, 173, 173, 0.637);
  }
  50% {
    box-shadow: 0 0 0 7px rgba(17, 221, 221, 0.493);
  }
  100% {
    box-shadow: 0 0 0 0 rgba(128, 255, 255, 0.61);
  }
}
