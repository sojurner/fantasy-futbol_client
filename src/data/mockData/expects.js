export const setCurrentPlayer = { playerId: 1, type: 'SET_CURRENT_PLAYER' };

export const setPlayerInfo = {
  player: {
    Acceleration: '76',
    Age: '26',
    Aggression: '68',
    Agility: '80',
    Balance: '75',
    Ball_control: '87',
    Club: 'Manchester City',
    Club_Logo: 'https://cdn.sofifa.org/24/18/teams/10.png',
    Composure: '84',
    Crossing: '90',
    Curve: '83',
    Dribbling: '85',
    Finishing: '83',
    Free_kick_accuracy: '83',
    GK_diving: '15',
    GK_handling: '13',
    GK_kicking: '5',
    GK_positioning: '10',
    GK_reflexes: '13',
    Heading_accuracy: '53',
    Interceptions: '56',
    Jumping: '65',
    Long_passing: '84',
    Long_shots: '86',
    Marking: '30',
    Name: 'K. De Bruyne',
    Nationality: 'Belgium',
    Overall: '89',
    Penalties: '77',
    Photo: 'https://cdn.sofifa.org/players/4/18/192985.png',
    Positioning: '84',
    Positions: 'RM CM CAM',
    Potential: '92',
    Reactions: '88',
    Short_passing: '90',
    Shot_power: '85',
    Sliding_tackle: '40',
    Sprint_speed: '75',
    Stamina: '87',
    Standing_tackle: '51',
    Strength: '73',
    Value: '€83M',
    Vision: '90',
    Volleys: '82',
    Wage: '€285K',
    country_id: 7,
    created_at: '2018-10-12T01:54:55.564Z',
    id: 7
  },
  type: 'SET_PLAYER_INFO'
};

export const setCurrentUser = { type: 'SET_CURRENT_USER', userId: 2 };

export const getPlayers = [
  {
    Name: 'L. Messi',
    id: 1,
    Photo: 'https://cdn.sofifa.org/players/4/18/158023.png',
    Nationality: 'Argentina',
    Positions: 'RW',
    Club: 'FC Barcelona',
    Overall: '93',
    Value: '€105M',
    Wage: '€565K'
  },
  {
    Name: 'Neymar',
    id: 2,
    Photo: 'https://cdn.sofifa.org/players/4/18/190871.png',
    Nationality: 'Brazil',
    Positions: 'LW',
    Club: 'Paris Saint-Germain',
    Overall: '92',
    Value: '€123M',
    Wage: '€280K'
  }
];

export const getCountries = [
  {
    id: 1,
    name: 'Argentina',
    flag: 'https://cdn.sofifa.org/flags/52.png',
    created_at: '2018-10-12T19:21:40.728Z'
  },
  {
    id: 2,
    name: 'Portugal',
    flag: 'https://cdn.sofifa.org/flags/38.png',
    created_at: '2018-10-12T19:21:40.728Z'
  },
  {
    id: 3,
    name: 'Brazil',
    flag: 'https://cdn.sofifa.org/flags/54.png',
    created_at: '2018-10-12T19:21:40.728Z'
  }
];

export const getPlayersByCountry = [
  {
    Name: 'C. Ninga',
    id: 2888,
    Photo: 'https://cdn.sofifa.org/players/4/18/231041.png',
    Nationality: 'Chad',
    Positions: 'LM ST',
    Club: 'Montpellier Hérault SC',
    Overall: '74',
    Value: '€7.5M',
    Wage: '€24K'
  },
  {
    Name: 'S. Altama',
    id: 12027,
    Photo: 'https://cdn.sofifa.org/players/4/18/200965.png',
    Nationality: 'Chad',
    Positions: 'CDM',
    Club: 'La Berrichonne de Châteauroux',
    Overall: '62',
    Value: '€325K',
    Wage: '€1K'
  }
];

export const getPlayer = {
  stats: [
    {
      Acceleration: 92,
      Aggression: 48,
      Agility: 90,
      Balance: 95,
      Composure: 96,
      Strength: 59,
      Sprint_speed: 87,
      Jumping: 68,
      Vision: 90,
      Stamina: 73
    },
    {
      Free_kick_accuracy: 90,
      Shot_power: 85,
      Long_shots: 88,
      Heading_accuracy: 71,
      Finishing: 95,
      Dribbling: 97,
      Ball_control: 95,
      Positioning: 93,
      Curve: 89
    },
    {
      Interceptions: 22,
      Marking: 13,
      Reactions: 95,
      Sliding_tackle: 26,
      Standing_tackle: 28,
      Crossing: 77,
      Volleys: 85,
      Short_passing: 88,
      Long_passing: 87
    },
    {
      GK_diving: 6,
      GK_handling: 11,
      GK_kicking: 15,
      GK_positioning: 14,
      GK_reflexes: 8
    }
  ],
  statType: ['Skills', 'Offense', 'Defense', 'Goal-Keeping'],
  info: { id: 1, Name: 'L. Messi' }
};

export const getPlayerByName = [
  {
    Name: 'L. Messi',
    id: 1,
    Photo: 'https://cdn.sofifa.org/players/4/18/158023.png',
    Nationality: 'Argentina',
    Positions: 'RW',
    Club: 'FC Barcelona',
    Overall: '93',
    Value: '€105M',
    Wage: '€565K'
  }
];

export const getPlayerByClub = [
  {
    Name: 'L. Reed',
    id: 10360,
    Photo: 'https://cdn.sofifa.org/players/4/18/222477.png',
    Nationality: 'England',
    Positions: 'CM',
    Club: 'Chesterfield',
    Overall: '63',
    Value: '€600K',
    Wage: '€5K'
  }
];

export const getPlayersByUser = [
  {
    id: 1,
    country_id: 1,
    Name: 'L. Messi',
    Age: '30',
    Photo: 'https://cdn.sofifa.org/players/4/18/158023.png',
    Club_Logo: 'https://cdn.sofifa.org/24/18/teams/241.png',
    Nationality: 'Argentina',
    Positions: 'RW',
    Club: 'FC Barcelona',
    Overall: '93',
    Potential: '93',
    Value: '€105M',
    Wage: '€565K',
    Acceleration: '92',
    Aggression: '48',
    Agility: '90',
    Balance: '95',
    Ball_control: '95',
    Composure: '96',
    Crossing: '77',
    Curve: '89',
    Dribbling: '97',
    Finishing: '95',
    Free_kick_accuracy: '90',
    GK_diving: '6',
    GK_handling: '11',
    GK_kicking: '15',
    GK_positioning: '14',
    GK_reflexes: '8',
    Heading_accuracy: '71',
    Interceptions: '22',
    Jumping: '68',
    Long_passing: '87',
    Long_shots: '88',
    Marking: '13',
    Penalties: '74',
    Positioning: '93',
    Reactions: '95',
    Short_passing: '88',
    Shot_power: '85',
    Sliding_tackle: '26',
    Sprint_speed: '87',
    Stamina: '73',
    Standing_tackle: '28',
    Strength: '59',
    Vision: '90',
    Volleys: '85',
    created_at: '2018-10-12T19:21:43.168Z'
  }
];
