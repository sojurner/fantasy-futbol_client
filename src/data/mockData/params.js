export const setPlayerInfo = {
  id: 7,
  country_id: 7,
  Name: 'K. De Bruyne',
  Age: '26',
  Photo: 'https://cdn.sofifa.org/players/4/18/192985.png',
  Club_Logo: 'https://cdn.sofifa.org/24/18/teams/10.png',
  Nationality: 'Belgium',
  Positions: 'RM CM CAM',
  Club: 'Manchester City',
  Overall: '89',
  Potential: '92',
  Value: '€83M',
  Wage: '€285K',
  Acceleration: '76',
  Aggression: '68',
  Agility: '80',
  Balance: '75',
  Ball_control: '87',
  Composure: '84',
  Crossing: '90',
  Curve: '83',
  Dribbling: '85',
  Finishing: '83',
  Free_kick_accuracy: '83',
  GK_diving: '15',
  GK_handling: '13',
  GK_kicking: '5',
  GK_positioning: '10',
  GK_reflexes: '13',
  Heading_accuracy: '53',
  Interceptions: '56',
  Jumping: '65',
  Long_passing: '84',
  Long_shots: '86',
  Marking: '30',
  Penalties: '77',
  Positioning: '84',
  Reactions: '88',
  Short_passing: '90',
  Shot_power: '85',
  Sliding_tackle: '40',
  Sprint_speed: '75',
  Stamina: '87',
  Standing_tackle: '51',
  Strength: '73',
  Vision: '90',
  Volleys: '82',
  created_at: '2018-10-12T01:54:55.564Z'
};

export const getPlayers = [
  'https://fantasy-futbol.herokuapp.com/api/v1/players?start=0&end=2'
];

export const getCountries = [
  'https://fantasy-futbol.herokuapp.com/api/v1/countries'
];

export const getPlayersByCountry = [
  'https://fantasy-futbol.herokuapp.com/api/v1/countries/100/players'
];

export const getPlayer = [
  'https://fantasy-futbol.herokuapp.com/api/v1/players/1/'
];

export const getPlayerByName = [
  'https://fantasy-futbol.herokuapp.com/api/v1/players?name=L. Messi'
];

export const getPlayersByClub = [
  'https://fantasy-futbol.herokuapp.com/api/v1/players?club=Chesterfield'
];

export const addUser = [
  'https://fantasy-futbol.herokuapp.com/api/v1/users',
  {
    body: '"Paul"',
    headers: { 'Content-Type': 'application/json' },
    method: 'POST'
  }
];

export const getUsers = ['https://fantasy-futbol.herokuapp.com/api/v1/users'];

export const getPlayersByUserParam = {
  id: 21,
  username: 'Paul',
  password: null,
  player_id_1: 1,
  player_id_2: null,
  player_id_3: null,
  player_id_4: null,
  player_id_5: null,
  player_id_6: null,
  player_id_7: null,
  player_id_8: null,
  player_id_9: null,
  player_id_10: null,
  player_id_11: null,
  created_at: '2018-10-17T11:38:03.733Z'
};

export const getPlayersByUser = [
  'https://fantasy-futbol.herokuapp.com/api/v1/players/1'
];

export const deleteUser = [
  'https://fantasy-futbol.herokuapp.com/api/v1/users/21',
  { method: 'DELETE' }
];
