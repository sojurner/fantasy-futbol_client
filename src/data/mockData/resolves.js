export const getAllPlayersResult = [
  {
    id: 1,
    country_id: 1,
    Name: 'L. Messi',
    Age: '30',
    Photo: 'https://cdn.sofifa.org/players/4/18/158023.png',
    Club_Logo: 'https://cdn.sofifa.org/24/18/teams/241.png',
    Nationality: 'Argentina',
    Positions: 'RW',
    Club: 'FC Barcelona',
    Overall: '93',
    Potential: '93',
    Value: '€105M',
    Wage: '€565K',
    Acceleration: '92',
    Aggression: '48',
    Agility: '90',
    Balance: '95',
    Ball_control: '95',
    Composure: '96',
    Crossing: '77',
    Curve: '89',
    Dribbling: '97',
    Finishing: '95',
    Free_kick_accuracy: '90',
    GK_diving: '6',
    GK_handling: '11',
    GK_kicking: '15',
    GK_positioning: '14',
    GK_reflexes: '8',
    Heading_accuracy: '71',
    Interceptions: '22',
    Jumping: '68',
    Long_passing: '87',
    Long_shots: '88',
    Marking: '13',
    Penalties: '74',
    Positioning: '93',
    Reactions: '95',
    Short_passing: '88',
    Shot_power: '85',
    Sliding_tackle: '26',
    Sprint_speed: '87',
    Stamina: '73',
    Standing_tackle: '28',
    Strength: '59',
    Vision: '90',
    Volleys: '85',
    created_at: '2018-10-12T19:21:43.168Z'
  },
  {
    id: 2,
    country_id: 3,
    Name: 'Neymar',
    Age: '25',
    Photo: 'https://cdn.sofifa.org/players/4/18/190871.png',
    Club_Logo: 'https://cdn.sofifa.org/24/18/teams/73.png',
    Nationality: 'Brazil',
    Positions: 'LW',
    Club: 'Paris Saint-Germain',
    Overall: '92',
    Potential: '94',
    Value: '€123M',
    Wage: '€280K',
    Acceleration: '94',
    Aggression: '56',
    Agility: '96',
    Balance: '82',
    Ball_control: '95',
    Composure: '92',
    Crossing: '75',
    Curve: '81',
    Dribbling: '96',
    Finishing: '89',
    Free_kick_accuracy: '84',
    GK_diving: '9',
    GK_handling: '9',
    GK_kicking: '15',
    GK_positioning: '15',
    GK_reflexes: '11',
    Heading_accuracy: '62',
    Interceptions: '36',
    Jumping: '61',
    Long_passing: '75',
    Long_shots: '77',
    Marking: '21',
    Penalties: '81',
    Positioning: '90',
    Reactions: '88',
    Short_passing: '81',
    Shot_power: '80',
    Sliding_tackle: '33',
    Sprint_speed: '90',
    Stamina: '78',
    Standing_tackle: '24',
    Strength: '53',
    Vision: '80',
    Volleys: '83',
    created_at: '2018-10-12T19:21:43.210Z'
  }
];

export const getCountriesResult = [
  {
    id: 1,
    name: 'Argentina',
    flag: 'https://cdn.sofifa.org/flags/52.png',
    created_at: '2018-10-12T19:21:40.728Z'
  },
  {
    id: 2,
    name: 'Portugal',
    flag: 'https://cdn.sofifa.org/flags/38.png',
    created_at: '2018-10-12T19:21:40.728Z'
  },
  {
    id: 3,
    name: 'Brazil',
    flag: 'https://cdn.sofifa.org/flags/54.png',
    created_at: '2018-10-12T19:21:40.728Z'
  }
];

export const getPlayersByCountryResult = [
  {
    id: 2888,
    country_id: 100,
    Name: 'C. Ninga',
    Age: '24',
    Photo: 'https://cdn.sofifa.org/players/4/18/231041.png',
    Club_Logo: 'https://cdn.sofifa.org/24/18/teams/70.png',
    Nationality: 'Chad',
    Positions: 'LM ST',
    Club: 'Montpellier Hérault SC',
    Overall: '74',
    Potential: '79',
    Value: '€7.5M',
    Wage: '€24K',
    Acceleration: '87',
    Aggression: '69',
    Agility: '73',
    Balance: '59',
    Ball_control: '69',
    Composure: '67',
    Crossing: '66',
    Curve: '65',
    Dribbling: '74',
    Finishing: '75',
    Free_kick_accuracy: '53',
    GK_diving: '6',
    GK_handling: '11',
    GK_kicking: '9',
    GK_positioning: '15',
    GK_reflexes: '9',
    Heading_accuracy: '63',
    Interceptions: '24',
    Jumping: '78',
    Long_passing: '55',
    Long_shots: '72',
    Marking: '20',
    Penalties: '72',
    Positioning: '72',
    Reactions: '71',
    Short_passing: '62',
    Shot_power: '79',
    Sliding_tackle: '23',
    Sprint_speed: '89',
    Stamina: '84',
    Standing_tackle: '25',
    Strength: '79',
    Vision: '59',
    Volleys: '75',
    created_at: '2018-10-12T19:21:47.879Z'
  },
  {
    id: 12027,
    country_id: 100,
    Name: 'S. Altama',
    Age: '26',
    Photo: 'https://cdn.sofifa.org/players/4/18/200965.png',
    Club_Logo: 'https://cdn.sofifa.org/24/18/teams/212.png',
    Nationality: 'Chad',
    Positions: 'CDM',
    Club: 'La Berrichonne de Châteauroux',
    Overall: '62',
    Potential: '65',
    Value: '€325K',
    Wage: '€1K',
    Acceleration: '63',
    Aggression: '71',
    Agility: '59',
    Balance: '63',
    Ball_control: '56',
    Composure: '66',
    Crossing: '47',
    Curve: '55',
    Dribbling: '56',
    Finishing: '39',
    Free_kick_accuracy: '55',
    GK_diving: '9',
    GK_handling: '13',
    GK_kicking: '7',
    GK_positioning: '15',
    GK_reflexes: '14',
    Heading_accuracy: '61',
    Interceptions: '58',
    Jumping: '75',
    Long_passing: '55',
    Long_shots: '35',
    Marking: '62',
    Penalties: '55',
    Positioning: '37',
    Reactions: '57',
    Short_passing: '58',
    Shot_power: '58',
    Sliding_tackle: '57',
    Sprint_speed: '67',
    Stamina: '84',
    Standing_tackle: '66',
    Strength: '73',
    Vision: '55',
    Volleys: '39',
    created_at: '2018-10-12T19:21:57.546Z'
  }
];

export const getPlayerResult = [
  {
    id: 1,
    country_id: 1,
    Name: 'L. Messi',
    Age: '30',
    Photo: 'https://cdn.sofifa.org/players/4/18/158023.png',
    Club_Logo: 'https://cdn.sofifa.org/24/18/teams/241.png',
    Nationality: 'Argentina',
    Positions: 'RW',
    Club: 'FC Barcelona',
    Overall: '93',
    Potential: '93',
    Value: '€105M',
    Wage: '€565K',
    Acceleration: '92',
    Aggression: '48',
    Agility: '90',
    Balance: '95',
    Ball_control: '95',
    Composure: '96',
    Crossing: '77',
    Curve: '89',
    Dribbling: '97',
    Finishing: '95',
    Free_kick_accuracy: '90',
    GK_diving: '6',
    GK_handling: '11',
    GK_kicking: '15',
    GK_positioning: '14',
    GK_reflexes: '8',
    Heading_accuracy: '71',
    Interceptions: '22',
    Jumping: '68',
    Long_passing: '87',
    Long_shots: '88',
    Marking: '13',
    Penalties: '74',
    Positioning: '93',
    Reactions: '95',
    Short_passing: '88',
    Shot_power: '85',
    Sliding_tackle: '26',
    Sprint_speed: '87',
    Stamina: '73',
    Standing_tackle: '28',
    Strength: '59',
    Vision: '90',
    Volleys: '85',
    created_at: '2018-10-12T19:21:43.168Z'
  }
];

export const getPlayerByNameResult = [
  {
    id: 1,
    country_id: 1,
    Name: 'L. Messi',
    Age: '30',
    Photo: 'https://cdn.sofifa.org/players/4/18/158023.png',
    Club_Logo: 'https://cdn.sofifa.org/24/18/teams/241.png',
    Nationality: 'Argentina',
    Positions: 'RW',
    Club: 'FC Barcelona',
    Overall: '93',
    Potential: '93',
    Value: '€105M',
    Wage: '€565K',
    Acceleration: '92',
    Aggression: '48',
    Agility: '90',
    Balance: '95',
    Ball_control: '95',
    Composure: '96',
    Crossing: '77',
    Curve: '89',
    Dribbling: '97',
    Finishing: '95',
    Free_kick_accuracy: '90',
    GK_diving: '6',
    GK_handling: '11',
    GK_kicking: '15',
    GK_positioning: '14',
    GK_reflexes: '8',
    Heading_accuracy: '71',
    Interceptions: '22',
    Jumping: '68',
    Long_passing: '87',
    Long_shots: '88',
    Marking: '13',
    Penalties: '74',
    Positioning: '93',
    Reactions: '95',
    Short_passing: '88',
    Shot_power: '85',
    Sliding_tackle: '26',
    Sprint_speed: '87',
    Stamina: '73',
    Standing_tackle: '28',
    Strength: '59',
    Vision: '90',
    Volleys: '85',
    created_at: '2018-10-12T19:21:43.168Z'
  }
];

export const getPlayersByClubResult = [
  {
    id: 10360,
    country_id: 13,
    Name: 'L. Reed',
    Age: '19',
    Photo: 'https://cdn.sofifa.org/players/4/18/222477.png',
    Club_Logo: 'https://cdn.sofifa.org/24/18/teams/1924.png',
    Nationality: 'England',
    Positions: 'CM',
    Club: 'Chesterfield',
    Overall: '63',
    Potential: '74',
    Value: '€600K',
    Wage: '€5K',
    Acceleration: '69',
    Aggression: '72',
    Agility: '84',
    Balance: '84',
    Ball_control: '68',
    Composure: '63',
    Crossing: '44',
    Curve: '39',
    Dribbling: '64',
    Finishing: '46',
    Free_kick_accuracy: '57',
    GK_diving: '9',
    GK_handling: '7',
    GK_kicking: '10',
    GK_positioning: '12',
    GK_reflexes: '12',
    Heading_accuracy: '42',
    Interceptions: '58',
    Jumping: '72',
    Long_passing: '64',
    Long_shots: '50',
    Marking: '45',
    Penalties: '43',
    Positioning: '54',
    Reactions: '53',
    Short_passing: '67',
    Shot_power: '52',
    Sliding_tackle: '59',
    Sprint_speed: '68',
    Stamina: '67',
    Standing_tackle: '54',
    Strength: '54',
    Vision: '60',
    Volleys: '37',
    created_at: '2018-10-12T19:21:55.767Z'
  }
];

export const getUsersResult = [
  {
    id: 21,
    username: 'Paul',
    password: null,
    player_id_1: null,
    player_id_2: null,
    player_id_3: null,
    player_id_4: null,
    player_id_5: null,
    player_id_6: null,
    player_id_7: null,
    player_id_8: null,
    player_id_9: null,
    player_id_10: null,
    player_id_11: null,
    created_at: '2018-10-17T11:38:03.733Z'
  },
  {
    id: 20,
    username: 'Ben',
    password: null,
    player_id_1: 1,
    player_id_2: 4,
    player_id_3: 2,
    player_id_4: 11,
    player_id_5: 2,
    player_id_6: 10,
    player_id_7: null,
    player_id_8: null,
    player_id_9: null,
    player_id_10: null,
    player_id_11: null,
    created_at: '2018-10-17T02:19:27.976Z'
  }
];

const getPlayersByUserResult = [
  [
    {
      id: 1,
      country_id: 1,
      Name: 'L. Messi',
      Age: '30',
      Photo: 'https://cdn.sofifa.org/players/4/18/158023.png',
      Club_Logo: 'https://cdn.sofifa.org/24/18/teams/241.png',
      Nationality: 'Argentina',
      Positions: 'RW',
      Club: 'FC Barcelona',
      Overall: '93',
      Potential: '93',
      Value: '€105M',
      Wage: '€565K',
      Acceleration: '92',
      Aggression: '48',
      Agility: '90',
      Balance: '95',
      Ball_control: '95',
      Composure: '96',
      Crossing: '77',
      Curve: '89',
      Dribbling: '97',
      Finishing: '95',
      Free_kick_accuracy: '90',
      GK_diving: '6',
      GK_handling: '11',
      GK_kicking: '15',
      GK_positioning: '14',
      GK_reflexes: '8',
      Heading_accuracy: '71',
      Interceptions: '22',
      Jumping: '68',
      Long_passing: '87',
      Long_shots: '88',
      Marking: '13',
      Penalties: '74',
      Positioning: '93',
      Reactions: '95',
      Short_passing: '88',
      Shot_power: '85',
      Sliding_tackle: '26',
      Sprint_speed: '87',
      Stamina: '73',
      Standing_tackle: '28',
      Strength: '59',
      Vision: '90',
      Volleys: '85',
      created_at: '2018-10-12T19:21:43.168Z'
    }
  ],
  null,
  null,
  null,
  null,
  null,
  null,
  null,
  null,
  null,
  null
];
