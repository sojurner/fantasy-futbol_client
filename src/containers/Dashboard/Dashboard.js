import React, { Component } from 'react';
import { connect } from 'react-redux';
import { NavLink, BrowserRouter } from 'react-router-dom';
import Modal from 'react-responsive-modal';

import { setCurrentUser } from '../../actions/userActions';
import * as fetch from '../../helpers/apiCalls/apiCalls';

import './Dashboard.css';

export class Dashboard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      open: false,
      users: [],
      usersPlayers: [],
      newUserInput: '',
      message: {}
    };
  }

  componentDidMount() {
    this.setState({ message: this.props.message });
    this.getUsers();
  }

  getUsers = async () => {
    const users = await fetch.getUsers();
    this.setState({ users });
  };

  onCloseModal = () => {
    this.setState({ open: false });
  };

  handleChange = event => {
    event.preventDefault();
    const { value } = event.target;
    this.setState({ newUserInput: value });
  };

  addUser = () => {
    this.setState({ open: true });
  };

  createUser = async e => {
    e.preventDefault();
    const { newUserInput, users } = this.state;
    const newUser = {
      username: newUserInput
    };
    await fetch.addUser(newUser);
    this.setState({
      users: [...users, newUser]
    });
  };

  deleteUser = async id => {
    await fetch.deleteUser(id);
    this.getUsers();
  };

  // deletePlayer = async playerIndex => {
  //   await fetch.addPlayerToUser = async (userId, playerId)
  // }

  displayUsers = () => {
    return this.state.users.map((user, index) => {
      return (
        <div key={`user-${index}`} className="user-option">
          <NavLink
            to={`/user${index}`}
            onClick={this.getUsersPlayers}
            className="user"
          >
            {user.username}
          </NavLink>
					
          <i
            onClick={() => this.deleteUser(user.id)}
            className="fas fa-minus-circle"
          />
        </div>
      );
    });
  };

  displayUsersPlayers = () => {
    return this.state.usersPlayers.map((player, index) => {
      return (
        <div className="dashboard-players">
          <p className="dashboard-players-playername">
            {player.Name} 
          </p>
          <i
            onClick={() => this.deletePlayer(index)}
            class="fas fa-minus-circle"
          />
        </div>
      );
    });
  };

  getUsersPlayers = async e => {
    const { textContent } = e.target;
    const matchedUser = this.state.users.find(
      user => user.username === textContent
    );
    this.props.addUser(matchedUser.id);
    const players = await fetch.getPlayersByUser(matchedUser);
    this.setState({ usersPlayers: players });
  };

  render() {
    const users = this.displayUsers();
    const usersPlayers = this.displayUsersPlayers();
    const { player } = this.props;
    return (
      <BrowserRouter>
        <section className="dashboard">
          <div>
            <div className="user-icon">
              <h3 className="users">Users</h3>
              <i className="fas fa-plus-circle" onClick={this.addUser} />
            </div>
            <nav className="user-nav">{users}</nav>
          </div>
          <div>
            <div className="user-icon">
              <h3 className="users">Team</h3>
              <i className="fas fa-plus-circle" />
            </div>
            <nav className="user-nav">
              {this.state.usersPlayers !== [] && usersPlayers}
              {player !== {} && (
								<div className="dashboard-players">
								<p className="dashboard-players-playername">
									{player.Name} 
								</p>
								<i onClick={() => this.deletePlayer()} class="fas fa-minus-circle" />
							</div>
              )}
            </nav>
          </div>

          <Modal
            open={this.state.open}
            onClose={this.onCloseModal}
            className="modal"
            center
          >
            <form onSubmit={this.createUser}>
              <input
                autoFocus
                name="username"
                value={this.state.newUser}
                className="username-input"
                placeholder="Enter Username"
                onChange={this.handleChange}
              />
              <button className="username-button">Create User</button>
            </form>
          </Modal>
        </section>
      </BrowserRouter>
    );
  }
}

const mapStateToProps = state => ({
  user: state.user,
  playerId: state.player.id,
  player: state.player.info
});

const mapDispatchToProps = dispatch => ({
  addUser: user => dispatch(setCurrentUser(user))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Dashboard);
