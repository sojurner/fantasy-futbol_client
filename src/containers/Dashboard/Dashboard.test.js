import React from 'react';
import { Dashboard } from './Dashboard';
import * as params from '../../data/mockData/params';

// describe('Dashboard', () => {
//   let wrapper;
//   let mockUser;
//   let mockPlayerId;
//   let mockPlayerInfo;
//   let mockAddUser;

//   beforeEach(() => {
//     mockPlayerId = '';
//     mockPlayerInfo = params.setPlayerInfo;
//     mockUser = '';
//     mockAddUser = jest.fn();

//     wrapper = shallow(
//       <Dashboard
//         user={mockUser}
//         playerId={mockPlayerId}
//         player={mockPlayerInfo}
//         addUser={mockAddUser}
//       />
//     );
//   });

//   it('should matchsnapshot', () => {
//     expect(wrapper).toMatchSnapshot();
//   });
// });
