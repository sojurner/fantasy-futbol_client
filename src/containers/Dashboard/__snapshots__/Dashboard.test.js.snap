// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should matchsnapshot 1`] = `undefined`;
